SwiftGit2/Credentials.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Credentials.swift
+//  SwiftGit2
+//
+//  Created by Tom Booth on 29/02/2016.
+//  Copyright © 2016 GitHub, Inc. All rights reserved.
+//
+
+import Foundation
+import Result
+
+/// A Swift mapping of git_credtype_t
+public struct CredentialType : OptionSetType {
```

Comment:
I'm struggling to fully understand the credential API from libgit2 (`git_cred` in particular), but I don't like how closely the credentials stuff here mirrors the C API. I'd like to aim for a more native Swift API and let SwiftGit2 handle the translation to the C API.

So, for example, SwiftGit2 might do something like this:

```swift
public enum Credentials {
    case Plaintext(username: String, password: String)
    case SSHMemory(username: String, publicKey: String, privateKey: String, passphrase: String)
    ....

    private var credtype: git_credtype_t { ... }
}
```

But I'm not entirely sure how workable that is because I don't fully understand the libgit2 API. I'd love for you to either pursue a more native Swift API or educate me about why that wouldn't work.